llvm/test/Transforms/SandboxVectorizer/regions-from-metadata.ll

```diff
@@ -0,0 +1,16 @@
+; RUN: opt -disable-output --passes=sandbox-vectorizer \
+; RUN:    -sbvec-passes=print-instruction-count \
+; RUN:    -sbvec-use-regions-from-metadata %s | FileCheck %s
+
+define i8 @foo(i8 %v0, i8 %v1) {
+  %t0 = add i8 %v0, 1, !sandboxvec !0
```

Comment:
can you swap the regions on `%t0` and `%t1` just as an extra test to make sure `!1` is still counted as 2 instructions?

Reply:
Instructions in a function are scanned in order and every time we reach an instruction with not-yet-seen-before region node, we create the region for it. If we swap the regions like this (if I understood the comment correctly):
```
define i8 @foo(i8 %v0, i8 %v1) {
  %t0 = add i8 %v0, 1, !sandboxvec !1
  %t1 = add i8 %t0, %v1, !sandboxvec !0
  %t2 = add i8 %t1, %v1, !sandboxvec !1
  ret i8 %t2
}
!0 = distinct !{!"sandboxregion"}
!1 = distinct !{!"sandboxregion"}
```
We'll see
```
InstructionCount: 2
InstructionCount: 1
```
(region `!1` is created first with `%t0` and pushed to the result vector, then `!0` is created with `%t1`, then `%t2` gets added to `!1`).

Reply:
the reason I asked for this was to make sure `std::distance(R.begin(), R.end())` wasn't checking how far apart instructions were, but rather how many instructions are in the region. if that's already tested somewhere else then ignore my comment

Reply:
llvm/unittests/SandboxIR/RegionTest.cpp has some tests that check the iterator range, but I don't think we have one with a non-contiguous range like that. The iterator for a Region goes over the `SetVector` of instructions that were added to the Region, not actually traversing the IR, so it should work. I can add another test case to the unit test in a separate patch if you think it's worth it.

Reply:
having a small test to ensure it's iterating over the `SetVector` rather than somehow iterating over instructions would be good